src/aof.c

```diff
@@ -1083,10 +1083,21 @@ void flushAppendOnlyFile(int force) {
          * stop write commands before fsync called in one second,
          * the data in page cache cannot be flushed in time. */
         if (server.aof_fsync == AOF_FSYNC_EVERYSEC &&
-            server.aof_fsync_offset != server.aof_current_size &&
+            (server.aof_fsync_dirty ||
+            (!server.aof_fsync_dirty && server.aof_fsync_offset != server.aof_current_size)) &&
```

Comment:
i think we need it. like i said in https://github.com/redis/redis/pull/11973#discussion_r1148591889

after we done a AOFRW, the aof_current_size will be smaller, and then it will hit the `server.aof_fsync_offset != server.aof_current_size`. there are two cases:
1. we stop write before AOFRW, so everything is fsync-ed. then when we done a AOFRW, aof_current_size  become smaller, in this case, we don't need to do another fsync
2. The other is the one we're working on right now

Reply:
but if we return true from that `()` block when `aof_fsync_dirty` is true, `|| <something else>`, then that "something else is anyway only relevant if `aof_fsync_dirty` is false.
i.e. unless i'm missing something basic in boolean arithmetic (which sometimes happens), this expression excessive regardless of any theory about aofrw scenarios.

Reply:
ohh, i got it. yes, you are right